keyboards/shahnetwork/b/rules.mk

```diff
@@ -0,0 +1,30 @@
+# MCU things
+BOOTLOADER = stm32-dfu
+ARMV = 7
+MCU = cortex-m4
+MCU_FAMILY = STM32
+MCU_SERIES = STM32F4xx
+MCU_LDSCRIPT = STM32F412xE_EEPROM
+MCU_STARTUP = stm32f4xx
+BOARD = SHAH_STM32F412RET6
+USE_FPU = yes
+STM32_BOOTLOADER_ADDRESS = 0x1FFF0000
+EEPROM_DRIVER = wear_leveling
+WEAR_LEVELING_DRIVER = legacy
```

Comment:
The really proper way to support STM32F412xE would be fixing the support for them in the ChibiOS EFL driver (apparently the current code for STM32F412 supports only 1.5M and 1M flash sizes, but not 512K flash) and using the `embedded_flash` driver.